src/components/PopupWithForm.js

```diff
@@ -30,4 +31,12 @@ export default class PopupWithForm extends Popup {
     this._popupForm.reset();
     super.close();
   }
+
```

Comment:
Needs correction: please don't close the modal in the end of the submit handler. Сlose modal in the case of successful api response in `then` block in `index.js`, otherwise the form will close even if an error occurred and you will need to reopen the form and fill the values again.

Reply:
Needs correction: still needs correction (take a look at line 27)

Reply:
fixed!